src/java_bytecode/java_object_factory.cpp

```diff
@@ -216,32 +216,27 @@ exprt allocate_dynamic_object(
 /// and which takes care of creating the associated declarations.
 /// \param target_expr: expression to which the necessary memory will be
 ///   allocated
-/// \param allocate_type: type of the object allocated
 /// \param symbol_table: symbol table
 /// \param loc: location in the source
 /// \param output_code: code block to which the necessary code is added
-/// \param cast_needed: Boolean flags saying where we need to cast the malloc
-///   site
-/// \return Expression representing the malloc site allocated.
-exprt allocate_dynamic_object_with_decl(
+void allocate_dynamic_object_with_decl(
   const exprt &target_expr,
-  const typet &allocate_type,
   symbol_tablet &symbol_table,
   const source_locationt &loc,
-  code_blockt &output_code,
-  bool cast_needed)
+  code_blockt &output_code)
 {
   std::vector<const symbolt *> symbols_created;
-
   code_blockt tmp_block;
-  exprt result=allocate_dynamic_object(
+  const typet &allocate_type=target_expr.type().subtype();
```

Comment:
Is this the only functional change? It's hard to see there is a lot of refactoring in the same commit.